cmd/processor/app/processor.go

```diff
@@ -74,25 +74,29 @@ import (
 
 // Processor is responsible to process events
 type Processor struct {
-	logger                logger.Logger
-	functionLogger        logger.Logger
-	triggers              []trigger.Trigger
-	webAdminServer        *webadmin.Server
-	healthCheckServer     commonhealthcheck.Server
-	metricSinks           []metricsink.MetricSink
-	namedWorkerAllocators *worker.AllocatorSyncMap
-	eventTimeoutWatcher   *timeout.EventTimeoutWatcher
-	startComplete         bool
-	stop                  chan bool
+	logger                    logger.Logger
+	functionLogger            logger.Logger
+	triggers                  []trigger.Trigger
+	webAdminServer            *webadmin.Server
+	healthCheckServer         commonhealthcheck.Server
+	metricSinks               []metricsink.MetricSink
+	namedWorkerAllocators     *worker.AllocatorSyncMap
+	eventTimeoutWatcher       *timeout.EventTimeoutWatcher
+	startComplete             bool
+	stop                      chan bool
+	stopRestartTriggerRoutine chan bool
+	restartTriggerChan        chan trigger.Trigger
 }
 
 // NewProcessor returns a new Processor
 func NewProcessor(configurationPath string, platformConfigurationPath string) (*Processor, error) {
 	var err error
 
 	newProcessor := &Processor{
-		namedWorkerAllocators: worker.NewAllocatorSyncMap(),
-		stop:                  make(chan bool, 1),
+		namedWorkerAllocators:     worker.NewAllocatorSyncMap(),
+		stop:                      make(chan bool, 1),
+		stopRestartTriggerRoutine: make(chan bool, 1),
```

Comment:
we can use `stop` channel for `stopRestartTriggerRoutine`

Reply:
I don't agree, because `Start()` is blocked on the `stop` channel, see line 214:
```
	<-p.stop // Wait for stop
	p.logger.Info("Processor quitting")
```

If two routines wait on the same channel, the behavior is unstable. Which routine will stop and which won't?..

Reply:
gotcha.